sys/openbsd/sysctl.txt

```diff
@@ -0,0 +1,294 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <sys/types.h>
+include <sys/ucred.h>
+include <sys/mount.h>
+include <sys/socket.h>
+include <sys/sysctl.h>
+include <ddb/db_var.h>
+
+# CPU_CONSDEV
+# XXX can includes be made conditional based on the architecture?
+include <amd64/cpu.h>
+
+sysctl$ddb(name ptr[in, ctl_ddb], namelen len[name], oldp buffer[inout], oldlenp ptr[inout, len[oldp, int64]], newp buffer[in], newlen len[newp])
```

Comment:
I just realized that `namelen` is wrong here. `name` is just an int array pointer, `namelen` must denote the number of elements, not the number of bytes. So basically `sizeof(ctl_ddb) / 4`. Can that be expressed?

/cc @dvyukov

Reply:
```
"len": length of another field (for array it is number of elements), type-options:
	argname of the object
```
https://github.com/google/syzkaller/blob/master/docs/syscall_descriptions_syntax.md

Reply:
Thanks. However I'm not able to express what I want. Ideally, I would like an array type consisting of both constant entries and others referering to another `flag` type. Expressed in pseudo notation `array[const[CTL_DDB, int32], flags[ctl_ddb_flags, int32]]`. Or if `len` could accept an optional denominator like `len[name, 4]` but that might be too tailored for my need.

Do you have any other ideas? Or am I missing something?

Reply:
Oh, ok, I see.
We actually have precisely `bytesize4` which is sizeof/4. It won't work in general case, but will work here.
Another option would be: array of unions of const and flags (won't guarantee precise interleaving of const+flag), but `len` will work.